File: update.py
import pandas as pd
import urllib.parse
import os

def slim_safe_inject():
    csv_path = 'fc_template.csv'
    if not os.path.exists(csv_path):
        print(f"❌ 错误：找不到 {csv_path}")
        return
    
    # 1. 预先生成要插入的列表内容（存入内存，这部分很小）
    df = pd.read_csv(csv_path)
    df['id'] = df['id'].astype(str)
    FC_BASE_URL = "https://liming.me"
    
    # 2. 准备不同语言的内容
    content_cache = {}
    for lang in ['zh-TW', 'ja', 'en']:
        lines = []
        lang_title = "📂 範本清單" if lang == 'zh-TW' else "📂 テンプレートリスト"
        if lang == 'en': lang_title = "📂 Full Template List"
        lines.append(f"## {lang_title}\n")
        
        if lang == 'en':
            for l in ['zh-TW', 'ja']:
                lines.append(f"#### {l.upper()}\n")
                tmp_df = df[df['lang'] == l]
                for _, row in tmp_df.iterrows():
                    lines.append(f"- [{row['title']}]({FC_BASE_URL}/{row['lang']}/tpl/{row['code']})")
                lines.append("")
        else:
            tmp_df = df[df['lang'] == lang]
            for _, row in tmp_df.iterrows():
                lines.append(f"- **{row['id']}**: [{row['title']}]({FC_BASE_URL}/{row['lang']}/tpl/{row['code']})")
        
        content_cache[lang] = "\n".join(lines)

    # 3. 流式处理文件：逐行读取，避免 MemoryError
    files_to_update = {
        'README.md': 'en',
        'README.zh-TW.md': 'zh-TW',
        'README.ja.md': 'ja'
    }

    START_TAG = "<catelist>"
    END_TAG = "</catelist>"

    for filename, mode in files_to_update.items():
        if not os.path.exists(filename): continue
        
        temp_filename = filename + ".tmp"
        found_tags = False
        
        with open(filename, 'r', encoding='utf-8') as f_in, \
             open(temp_filename, 'w', encoding='utf-8') as f_out:
            
            skip_mode = False
            for line in f_in:
                if START_TAG in line:
                    f_out.write(line) # 写入开始标记
                    f_out.write("\n" + content_cache[mode] + "\n") # 写入新内容
                    skip_mode = True # 进入跳过模式，直到找到结束标记
                    found_tags = True
                    continue
                
                if END_TAG in line:
                    skip_mode = False # 结束标记找到了，恢复正常写入
                    f_out.write(line)
                    continue
                
                if not skip_mode:
                    f_out.write(line)

        if found_tags:
            os.replace(temp_filename, filename) # 用新文件替换旧文件
            print(f"✅ 流式更新成功: {filename}")
        else:
            os.remove(temp_filename)
            print(f"ℹ️ {filename} 未发现标记或文件损坏。")

if __name__ == "__main__":
    slim_safe_inject()

File: update.js
const fs = require('fs').promises;
const fsSync = require('fs');
const path = require('path');
const csv = require('csv-parser');

// 主函数
async function slimSafeInject() {
  const csvPath = 'fc_template.csv';

  // ====================== 在这里配置 ID 范围 ======================
  const START_ID = 0;  // 0 = 不限制起始
  const END_ID = 0;    // 0 = 不限制结束
  // ===============================================================

  // 检查 CSV 文件是否存在
  if (!fsSync.existsSync(csvPath)) {
    console.log(`❌ 错误：找不到 ${csvPath}`);
    return;
  }

  // 1. 读取 CSV
  const df = [];
  await new Promise((resolve, reject) => {
    fsSync.createReadStream(csvPath)
      .pipe(csv())
      .on('data', (row) => {
        df.push(row);
      })
      .on('end', resolve)
      .on('error', reject);
  });

  const FC_BASE_URL = 'https://liming.me';

  // 2. 生成多语言内容缓存
  const contentCache = {};
  const langs = ['zh-TW', 'ja', 'en'];

  for (const lang of langs) {
    const lines = [];
    let langTitle;

    if (lang === 'zh-TW') langTitle = '📂 範本清單';
    else if (lang === 'ja') langTitle = '📂 テンプレートリスト';
    else langTitle = '📂 Full Template List';

    lines.push(`## ${langTitle}\n`);

    if (lang === 'en') {
      for (const l of ['zh-TW', 'ja']) {
        lines.push(`#### ${l.toUpperCase()}\n`);
        const tmpDf = df.filter(row => {
          const matchLang = row.lang === l;
          if (START_ID === 0 && END_ID === 0) return matchLang;
          const id = Number(row.id);
          return matchLang && id >= START_ID && id <= END_ID;
        });
        let num = 1;
        for (const row of tmpDf) {
          lines.push(`- **${num}**: [${row.title}](${FC_BASE_URL}/${row.lang}/tpl/${row.code})`);
          num++;
        }
        lines.push('');
      }
    } else {
      const tmpDf = df.filter(row => {
        const matchLang = row.lang === lang;
        if (START_ID === 0 && END_ID === 0) return matchLang;
        const id = Number(row.id);
        return matchLang && id >= START_ID && id <= END_ID;
      });
      let num = 1;
      for (const row of tmpDf) {
        lines.push(`- **${num}**: [${row.title}](${FC_BASE_URL}/${row.lang}/case/${row.slug})`);
        num++;
      }
    }

    contentCache[lang] = lines.join('\n');
  }

  // 3. 更新 README 文件 + 自动删除标记
  const filesToUpdate = {
    'README.md': 'en',
    'README.zh-TW.md': 'zh-TW',
    'README.ja.md': 'ja'
  };

  const START_TAG = '<catelist>';
  const END_TAG = '</catelist>';

  for (const [filename, mode] of Object.entries(filesToUpdate)) {
    if (!fsSync.existsSync(filename)) continue;

    const tempFilename = filename + '.tmp';
    let foundTags = false;
    let skipMode = false;

    const content = await fs.readFile(filename, 'utf8');
    const lines = content.split(/\r?\n/);
    const output = [];

    for (const line of lines) {
      // 遇到开始标记：只插入内容，不保留标记本身
      if (line.includes(START_TAG)) {
        output.push(contentCache[mode]);
        skipMode = true;
        foundTags = true;
        continue;
      }

      // 遇到结束标记：跳过，不保留标记
      if (line.includes(END_TAG)) {
        skipMode = false;
        continue;
      }

      if (!skipMode) {
        output.push(line);
      }
    }

    await fs.writeFile(tempFilename, output.join('\n'), 'utf8');

    if (foundTags) {
      fsSync.renameSync(tempFilename, filename);
      console.log(`✅ 更新完成（已自动清除标记）: ${filename}`);
    } else {
      fsSync.unlinkSync(tempFilename);
      console.log(`ℹ️ ${filename} 未发现标记`);
    }
  }
}

// 运行
(async () => {
  try {
    await slimSafeInject();
    console.log('\🎉 全部任务完成！');
  } catch (err) {
    console.error('❌ 执行失败:', err);
  }
})();